File: app/src/main/java/com/coppermobile/mysampleusingmvvmlivedata/adapters/DishAdapter.java
package com.coppermobile.mysampleusingmvvmlivedata.adapters;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v7.util.DiffUtil;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.coppermobile.mysampleusingmvvmlivedata.R;
import com.coppermobile.mysampleusingmvvmlivedata.data.Dish;
import com.coppermobile.mysampleusingmvvmlivedata.utils.IClickListener;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

import butterknife.BindView;
import butterknife.ButterKnife;

public class DishAdapter extends RecyclerView.Adapter<DishAdapter.MyViewHolder> {

    private List<Dish> dishList;
    private Context context;
    private IClickListener iClickListener;

    public DishAdapter(Context context, IClickListener iClickListener) {
        this.context = context;
        this.iClickListener = iClickListener;
    }

    public void addData(List<Dish> dishList) {
        this.dishList = dishList;
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.list_dish, parent, false);
        return new MyViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull final MyViewHolder holder, int position) {
        final Dish dish = dishList.get(position);
        if (dish != null) {
            Glide.with(context).load(dish.getDishImage()).into(holder.ivDish);
            holder.tvName.setText(dish.getDishName());
            holder.tvDescription.setText(dish.getDishDescription());
            holder.tvPrice.setText(String.format("₹%s", String.valueOf(dish.getDishPrice())));
        }

        holder.rlContainer.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                holder.itemView.setBackgroundResource(R.drawable.item_clicked);
                if (iClickListener != null) {
                    iClickListener.onItemPressed(dish);
                }
            }
        });
    }

    @Override
    public int getItemCount() {
        return dishList != null && !dishList.isEmpty() ? dishList.size() : 0;
    }

    static class MyViewHolder extends RecyclerView.ViewHolder {

        @BindView(R.id.iv_dish)
        ImageView ivDish;
        @BindView(R.id.tv_name)
        TextView tvName;

        @BindView(R.id.tv_description)
        TextView tvDescription;
        @BindView(R.id.tv_price)
        TextView tvPrice;

        @BindView(R.id.rl_container)
        RelativeLayout rlContainer;

        MyViewHolder(View v) {
            super(v);
            ButterKnife.bind(this, v);
        }
    }
}

File: app/src/main/java/com/coppermobile/kotlinsamplemvvmlivedataroom/adapters/DishAdapter.kt
package com.coppermobile.kotlinsamplemvvmlivedataroom.adapters

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.RelativeLayout
import android.widget.TextView
import butterknife.BindView
import butterknife.ButterKnife
import com.bumptech.glide.Glide
import com.coppermobile.kotlinsamplemvvmlivedataroom.R
import com.coppermobile.kotlinsamplemvvmlivedataroom.data.Dish
import com.coppermobile.kotlinsamplemvvmlivedataroom.utils.IClickListener

class DishAdapter(private val context: Context, private val iClickListener: IClickListener?) : RecyclerView.Adapter<DishAdapter.MyViewHolder>() {

    private var dishList: List<Dish>? = null

    fun addData(dishList: List<Dish>?) {
        this.dishList = dishList
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        return MyViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.list_dish, parent, false))
    }

    override fun getItemCount(): Int {
        return dishList?.size ?: 0
    }


    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        val dish = dishList?.get(position)
        if (dish != null) {
            holder.tvName.text = dish.dishName
            holder.tvDescription.text = dish.dishDescription
            holder.tvPrice.text = String.format("₹%s", dish.dishPrice)
            Glide.with(context).load(dish.dishImage).into(holder.ivDish)

            holder.rlContainer.setOnClickListener {
                holder.itemView.setBackgroundResource(R.drawable.item_clicked)
                iClickListener?.onItemPressed(dish)
            }
        }
    }

    class MyViewHolder(v: View) : RecyclerView.ViewHolder(v) {

        @BindView(R.id.iv_dish)
        lateinit var ivDish: ImageView
        @BindView(R.id.tv_name)
        lateinit var tvName: TextView

        @BindView(R.id.tv_description)
        lateinit var tvDescription: TextView
        @BindView(R.id.tv_price)
        lateinit var tvPrice: TextView

        @BindView(R.id.rl_container)
        lateinit var rlContainer: RelativeLayout

        init {
            ButterKnife.bind(this, v)
        }
    }
}